src/Features/LanguageServer/Microsoft.CommonLanguageServerProtocol.Framework/HandlerProvider.cs

```diff
@@ -17,31 +17,31 @@ namespace Microsoft.CommonLanguageServerProtocol.Framework;
 internal class HandlerProvider : IHandlerProvider
 {
     private readonly ILspServices _lspServices;
-    private ImmutableDictionary<RequestHandlerMetadata, Lazy<IMethodHandler>>? _requestHandlers;
+    private ImmutableDictionary<RequestHandlerMetadata, List<Lazy<IMethodHandler, string?>>>? _requestHandlers;
 
     public HandlerProvider(ILspServices lspServices)
     {
         _lspServices = lspServices;
     }
 
     /// <summary>
-    /// Get the MethodHandler for a particular request.
+    /// Get the <see cref="IMethodHandler"/>s for a particular request.
     /// </summary>
     /// <param name="method">The method name being made.</param>
     /// <param name="requestType">The requestType for this method.</param>
     /// <param name="responseType">The responseType for this method.</param>
     /// <returns>The handler for this request.</returns>
-    public IMethodHandler GetMethodHandler(string method, Type? requestType, Type? responseType)
+    public ImmutableArray<Lazy<IMethodHandler, string?>> GetMethodHandlers(string method, Type? requestType, Type? responseType)
```

Comment:
Unfortunately we may not be able to change any public methods in this project like this.  Razor and webtools depend on these types :(

The changes need to be backwards compatible.

cc @ToddGrun

Reply:
We can audit the usages of these to figure out which methods and types are actually used.  I'm not sure if HandlerProvider is, but the interface is likely used.

We may need to introduce 2 versions of these.

Reply:
See my comment below, but IMHO this would be better if it could take in the language metadata as a string and return the singular correct handler

Reply:
Reverted to previous signature with optional language param